lib/config/optionsconfiguration.go

```diff
@@ -184,15 +186,35 @@ func (opts OptionsConfiguration) StunServers() []string {
 	for _, addr := range opts.RawStunServers {
 		switch addr {
 		case "default":
-			defaultPrimaryAddresses := make([]string, len(DefaultPrimaryStunServers))
-			copy(defaultPrimaryAddresses, DefaultPrimaryStunServers)
-			rand.Shuffle(defaultPrimaryAddresses)
-			addresses = append(addresses, defaultPrimaryAddresses...)
+			_, records, err := net.LookupSRV("stun", "udp", "syncthing.net")
+			if err == nil {
+				var priorities []uint16
+				priorityMap := make(map[uint16][]string)
+				for _, record := range records {
+					var priority = record.Priority
+					var target, _ = strings.CutSuffix(record.Target, ".")
+					var address = fmt.Sprintf("%s:%d", target, record.Port)
+					if priorityGroup, ok := priorityMap[priority]; ok {
+						priorityMap[priority] = append(priorityGroup, address)
+					} else {
+						priorityMap[priority] = []string{address}
+						priorities = append(priorities, priority)
+					}
+				}
+				slices.Sort(priorities)
+				for _, priority := range priorities {
+					var priorityGroup = priorityMap[priority]
+					rand.Shuffle(priorityGroup)
```

Comment:
The SRV record also offers a weight which we could use for a weighted shuffle. But I don't see a use case for that.